src/components/button/Button.js

```diff
@@ -0,0 +1,50 @@
+import PropTypes from "prop-types";
+import tw from "twin.macro";
+import theme from "@/themes/theme";
+import styled from "styled-components";
+
+const StyledButton = styled.button(
+  ({ className, styles: { block, color, border, shape, size } = {} }) => [
```

Comment:
Can you explain to me where `styles` comes from.  I understand that you are destructuring `block, color, border...` - but I don't understand with the `styles` object comes from.

Reply:
I put `block, color, border..` to one prop `styles`, so when you use `<Button>` you need put them in `styles`:
```
<Button
    className="test"
    styles={{
         border: "rounded",
         color: "primary
    }}
>Text</Button>
```
<img width="590" alt="Screenshot 2021-01-19 at 17 42 42" src="https://user-images.githubusercontent.com/29330551/105057291-c0fab600-5a7d-11eb-8dc3-59a03223aaf2.png">

Reply:
Got it.  

I personally would prefer NOT to put them all inside of `styles`.  I would rather just use each one independently -- something like this:

```
<Button color="primary" shape="rounded">Click Here</Button>
```

Is there an advantage to putting it all in `styles` -- or is it just a matter of preference?